### PR TITLE
Jhchen/ft manage account

### PR DESCRIPTION
**Pull request by ChenJiehan318**
1. Add add command for account
2. Add delete command for account
3. Add list command for account
4. Add parsers for account commands

### CODE REVIEW

File: src/main/java/budgetbuddy/MainApp.java [hunk 1]
```diff
@@ -76,14 +77,20 @@ public void init() throws Exception {
...
+        accountsManager = new AccountsManager();
+        // TODO Load loan data from storage (if data present).
         try {
             addressBookOptional = storage.readAddressBook();
             if (!addressBookOptional.isPresent()) {
                 logger.info("Data file not found. Will be starting with a sample Budget Buddy");
             }
+            /**if (!accountBookOptional.isPresent()) {
+                logger.info("Data file not found. Will be starting with a sample Budget Buddy");
+            }*/
```
Comment (thwnus) [inline src/main/java/budgetbuddy/MainApp.java]: Duplicate of the previous 3 lines?

File: src/main/java/budgetbuddy/model/AccountsManager.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        }
+    }
+
+
+
+    /**
+     * Deletes an account.
+     * @param toDelete The target account for deletion.
+     */
+    public void deleteAccount (Account toDelete) {
```
Comment (thwnus) [inline src/main/java/budgetbuddy/model/AccountsManager.java]: Stray space here (does this pass Checkstyle? o_O)

File: src/main/java/budgetbuddy/model/ModelManager.java [hunk 3]
```diff
@@ -28,20 +29,22 @@
...
-    public ModelManager(LoansManager loansManager, ReadOnlyAddressBook addressBook, ReadOnlyUserPrefs userPrefs) {
+    public ModelManager(LoansManager loansManager, AccountsManager accountsManager, ReadOnlyAddressBook addressBook,
+                        ReadOnlyUserPrefs userPrefs) {
         super();
         requireAllNonNull(loansManager, addressBook, userPrefs);
 
         logger.fine("Initializing with address book: " + addressBook + " and user prefs " + userPrefs);
 
         this.loansManager = new LoansManager(loansManager.getPersonsList());
+        this.accountsManager = new AccountsManager(accountsManager.getAccountsList());
```
Comment (thwnus) [inline src/main/java/budgetbuddy/model/ModelManager.java]: This is really strange. Why create a new AccountsManager when one is passed in?

File: src/main/java/budgetbuddy/MainApp.java (hunk 1 above)
Comment (ChenJiehan318) [reply src/main/java/budgetbuddy/MainApp.java]: Excellent observation. They are already deleted :)

File: src/main/java/budgetbuddy/model/AccountsManager.java (hunk 2 above)
Comment (ChenJiehan318) [reply src/main/java/budgetbuddy/model/AccountsManager.java]: Deleted. (yes)

File: src/main/java/budgetbuddy/model/ModelManager.java (hunk 3 above)
Comment (ChenJiehan318) [reply src/main/java/budgetbuddy/model/ModelManager.java]: yeah... but the other variables are also created this way, so i just followed it for consistency

File: src/main/java/budgetbuddy/model/account/UniqueAccountList.java [hunk 4]
```diff
@@ -75,6 +83,29 @@ public void remove(Account toRemove) {
...
+
+        Account targetAccount = null;
+        for (Account account : internalUnmodifiableList) {
+            if (account.isSameAccount(toGet)) {
+                targetAccount = account;
+            }
+        }
+
+        if (targetAccount == null) {
+            throw new PersonNotFoundException();
```
Comment (tcgj) [inline src/main/java/budgetbuddy/model/account/UniqueAccountList.java]: I think this is throwing the wrong exception

File: src/test/java/budgetbuddy/logic/commands/AddCommandIntegrationTest.java [hunk 5]
```diff
@@ -7,13 +7,15 @@
...
 
+import budgetbuddy.model.AccountsManager;
 import budgetbuddy.model.LoansManager;
 import budgetbuddy.model.Model;
 import budgetbuddy.model.ModelManager;
 import budgetbuddy.model.UserPrefs;
 import budgetbuddy.model.person.Person;
 import budgetbuddy.testutil.PersonBuilder;
 
+
```
Comment (tcgj) [inline src/test/java/budgetbuddy/logic/commands/AddCommandIntegrationTest.java]: Is this a stray empty line?

File: src/test/java/budgetbuddy/logic/commands/DeleteCommandTest.java [hunk 6]
```diff
@@ -13,6 +13,8 @@
 
 import budgetbuddy.commons.core.Messages;
 import budgetbuddy.commons.core.index.Index;
+
```
Comment (tcgj) [inline src/test/java/budgetbuddy/logic/commands/DeleteCommandTest.java]: Is this a stray empty line?

File: src/test/java/budgetbuddy/logic/commands/EditCommandTest.java [hunk 7]
```diff
@@ -26,13 +27,13 @@
 import budgetbuddy.model.person.Person;
 import budgetbuddy.testutil.EditPersonDescriptorBuilder;
 import budgetbuddy.testutil.PersonBuilder;
-
```
Comment (tcgj) [inline src/test/java/budgetbuddy/logic/commands/EditCommandTest.java]: Is this missing an empty line?

**Pull request merged**